bot/exts/moderation/clean.py

```diff
@@ -0,0 +1,580 @@
+        await self._clean_messages(ctx, traverse, bots_only=True, channels=channels, use_cache=use_cache)
+
+    @clean_group.command(name="regex", aliases=["word", "expression", "pattern"])
+    async def clean_regex(
+        self,
+        ctx: Context,
+        regex: Regex,
+        traverse: Optional[int] = DEFAULT_TRAVERSE,
+        use_cache: Optional[bool] = True,
+        *,
+        channels: CleanChannels = None
+    ) -> None:
+        """
+        Delete all messages that match a certain regex, stop cleaning after traversing `traverse` messages.
+
+        The pattern must be provided enclosed in backticks.
+        If the pattern contains spaces, and still needs to be enclosed in double quotes on top of that.
+        For example: `[0-9]`
+        """
+        await self._clean_messages(ctx, traverse, regex=regex, channels=channels, use_cache=use_cache)
+
+    @clean_group.command(name="until")
+    async def clean_until(
+            self,
+            ctx: Context,
+            until: CleanLimit,
+            channel: TextChannel = None
+    ) -> None:
+        """
+        Delete all messages until a certain limit.
+
+        A limit can be either a message, and ISO date-time string, or a time delta.
+        If a message is specified, `channel` cannot be specified.
+        """
+        await self._clean_messages(
+            ctx,
+            CleanMessages.message_limit,
+            channels=[channel] if channel else None,
+            first_limit=until,
+        )
+
+    @clean_group.command(name="between", aliases=["after-until", "from-to"])
+    async def clean_between(
+            self,
+            ctx: Context,
+            first_limit: CleanLimit,
+            second_limit: CleanLimit,
+            channel: TextChannel = None
+    ) -> None:
+        """
+        Delete all messages within range.
+
+        The range is specified through two limits.
+        A limit can be either a message, and ISO date-time string, or a time delta.
+
+        If two messages are specified, they both must be in the same channel.
+        If a message is specified, `channel` cannot be specified.
+        """
+        await self._clean_messages(
+            ctx,
+            CleanMessages.message_limit,
+            channels=[channel] if channel else None,
+            first_limit=first_limit,
+            second_limit=second_limit,
+        )
+
+    @clean_group.command(name="stop", aliases=["cancel", "abort"])
+    async def clean_cancel(self, ctx: Context) -> None:
+        """If there is an ongoing cleaning process, attempt to immediately cancel it."""
+        self.cleaning = False
+
+        embed = Embed(
+            color=Colour.blurple(),
+            description="Clean interrupted."
+        )
+        delete_after = 10
+        if is_mod_channel(ctx.channel):
+            delete_after = None
+        await ctx.send(embed=embed, delete_after=delete_after)
```

Comment:
If there is no cleaning going on here, we should let them know and not convey false information that the clean command has been interrupted in between.

Reply:
Yeah that seems right.